src/main/java/org/openrewrite/java/migrate/lang/RefineSwitchCases.java

```diff
@@ -0,0 +1,147 @@
+                            if (statement instanceof J.Case) {
+                                J.Case case_ = (J.Case) statement;
+                                if (!(case_.getBody() instanceof J.Block)) {
+                                    return statement;
+                                }
+                                List<String> labelVars = case_.getCaseLabels().stream()
+                                        .filter(J.VariableDeclarations.class::isInstance)
+                                        .flatMap(it -> getVariablesCreated((J.VariableDeclarations) it).stream())
+                                        .map(J.Identifier::getSimpleName)
+                                        .collect(toList());
+                                if (labelVars.isEmpty()) {
+                                    return statement;
+                                }
+                                List<Statement> caseStatements = ((J.Block) case_.getBody()).getStatements();
+                                if (caseStatements.size() == 1 && caseStatements.get(0) instanceof J.If) {
+                                    J.If if_ = (J.If) caseStatements.get(0);
+                                    if (getConditionVariables(if_.getIfCondition().getTree()).stream()
+                                            .allMatch(conditionVariable -> labelVars.contains(conditionVariable.getSimpleName()))) {
+                                        // Replace case with multiple cases
+                                        return getCases(case_, if_);
+                                    }
+                                }
+                            }
+                            return statement;
+                        })));
+                return mappedSwitch == switch_ ? switch_ : autoFormat(mappedSwitch, ctx);
+            }
+
+            private List<J.Identifier> getConditionVariables(@Nullable Expression expression) {
+                if (expression instanceof J.Identifier) {
+                    return singletonList((J.Identifier) expression);
+                } else if (expression instanceof J.Binary) {
+                    J.Binary binary = (J.Binary) expression;
+                    List<J.Identifier> variables = new ArrayList<>();
+                    variables.addAll(getConditionVariables(binary.getLeft()));
+                    variables.addAll(getConditionVariables(binary.getRight()));
+                    return variables;
+                } else if (expression instanceof J.MethodInvocation) {
+                    J.MethodInvocation methodInvocation = (J.MethodInvocation) expression;
+                    List<J.Identifier> variables = new ArrayList<>();
+                    variables.addAll(getConditionVariables(methodInvocation.getSelect()));
+                    variables.addAll(getConditionVariables(methodInvocation.getArguments()));
```

Comment:
```suggestion
                    for (Expression arg : methodInvocation.getArguments()) {
                        variables.addAll(getConditionVariables(arg));
                    }
```

Way shorter, then you don't need that extra method 😉.